flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkConfParser.java

```diff
@@ -21,20 +21,23 @@
 import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
+import javax.annotation.Nullable;
 import org.apache.flink.configuration.ConfigOption;
 import org.apache.flink.configuration.ReadableConfig;
 import org.apache.iceberg.Table;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 
 class FlinkConfParser {
 
   private final Map<String, String> tableProperties;
   private final Map<String, String> options;
   private final ReadableConfig readableConfig;
 
-  FlinkConfParser(Table table, Map<String, String> options, ReadableConfig readableConfig) {
-    this.tableProperties = table.properties();
+  FlinkConfParser(
+      @Nullable Table table, Map<String, String> options, ReadableConfig readableConfig) {
+    this.tableProperties = table == null ? Maps.newHashMap() : table.properties();
```

Comment:
Why would the table be null?

Reply:
These configurations are read from only two places:

* Options
* Flink Configure

Reply:
@hililiwei I think you meant that currently Flink read configurations do NOT have anything from Iceberg table properties at all. Hence the hierarchy doesn't use table properties. But we can probably still pass in a valid table object for source too, right?

Reply:
Yes.

Reply:
Here I think it needs to be modified further, because scan context will eventually take some table properties for comparison. Take this example:

```
if (context.splitSize() != null) {
   refinedScan = refinedScan.option(TableProperties.SPLIT_SIZE, context.splitSize().toString());
}
```

So, I want to go a little further and replace scancontext

`    
refinedScan = refinedScan.option(TableProperties.SPLIT_SIZE, flinkReadConf.splitSize());
`
It no longer requires comparison.

Reply:
@hililiwei can we remove the `@Nullable` annotation? current implementation provides a valid Table object. we don't really need this annotation. seems unnecessary.

Reply:
Done.

Reply:
this line can also be reverted now that table is always an valid/non-null object.

Reply:
@hililiwei I am going to merge this PR. you can address this one-liner in the backport (1.14 and 1.16) PR.